### ISSUE TITLE
[bug] Simple conan package cannot be found on Artifactory

### ISSUE THREAD

**Issue by mcleary** [conan-io/conan#7346]
I am working on a conan package that is uploaded to an artifactory server running internally, however, `conan search` is returning no references to the packages stored in artifactory. I managed to write a sample `conanfile.py` that can reproduce this bug, please see the details below.

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18.04
  * Compiler+version: gcc 7.5.0
  * Conan version: 1.23.0
  * Python version: 3.6.9

### Steps to reproduce (Include if Applicable)

* Run `conan create .` with the following `conanfile.py`

```python
from conans import ConanFile, tools
import os

class ConanSample(ConanFile):
    name = "SampleProject"
    version = "0.1"
    description = "A Sample Project"
    settings = "os", "compiler", "build_type", "arch"
    generators = "cmake"
    options = {
        "flags": "ANY"
    }
    default_options = {
        "flags": None
    }
    default_user = "user"
    default_channel = "test"

    def configure(self):
        # If I remove the = sign the search work
        self.options.flags = "a=b"

    def source(self):
        with open(os.path.join(self.source_folder, "some_file.txt"), "w") as some_file:
            some_file.write("Something")

    def package(self):        
        self.copy(pattern="some_file.txt")
```

* Run `conan upload -r artifactory --all SampleProject/0.1@user/test`
* Run `conan search -r artifactory SampleProject/0.1@user/test`

This will return the following

```
Existing packages for recipe SampleProject/0.1@user/test:

Existing recipe in remote 'artifactory':

There are no packages for reference 'SampleProject/0.1@user/test', but package recipe found.
```

Where `artifactory` is the artifactory server that I have running

Now here is the interesting bit, if I remove the `=` sign in the `conanfile.py` `configure` method, the package can be found! I took this from an internal project that I am working with and I narrow it down to this `=` sign in the `configure` method.

Any clues on why this is happening?

**Comment by memsharded** [comment]
Hi @mcleary 

I think this is a bug that was solved in Artifactory some time ago. Can you please tell which Artifactory (CE, PRO) and which version are you using?

**Comment by mcleary** [comment]
If it is solved that would be great. Here is the version

```
Artifactory Community Edition for C/C++
6.16.0 rev 61600900
```

**Comment by memsharded** [comment]
Thanks for the info @mcleary 

I have verified this. This is indeed a bug, it fails in:

- Artifactory 6.16-6.20 => There are no packages.
- Artifactory 7.5 => ``ERROR: 500: org.jfrog.repomd.conan.exceptions.ConanInfoException: Invalid conaninfo.txt format: flags=a=b. [Remote: artifactory]``

Fortunately, it seems this doesn't affect to package resolution, package can still be used, installed, found in the web UI (the ConanInfo tab is empty because of this), so the severity seems to not be very high.  Assigned @czoido to follow up with Artifactory team (not a Conan client issue). Thanks very much for reporting and providing a detailed case to reproduce!

**Comment by mcleary** [comment]
Excellent! Glad you can reproduce it. As you said, the package can still be used, it is just quite annoying not being able to search for it.

**Comment by czoido** [comment]
The same problem reported here: https://github.com/conan-io/conan/issues/5642
Already reported in Artifactory JIRA: https://www.jfrog.com/jira/browse/RTFACT-19943

**Comment by czoido** [comment]
Closed as this should have been solved by Artifactory 7.12.3
